src/index.tsx

```diff
@@ -38,8 +38,8 @@ function useWavesurferInstance(
   options: Partial<WaveSurferOptions>,
 ): WaveSurfer | null {
   const [wavesurfer, setWavesurfer] = useState<WaveSurfer | null>(null)
-  // Flatten options object to an array of keys and values to compare them deeply in the hook deps
-  const flatOptions = useMemo(() => Object.entries(options).flat(), [options])
+  // Serialize options object to an array of keys and values to compare them deeply in the hook deps
+  const serialOptions = useMemo(() => Object.entries(options).flat().toString(), [options])
```

Comment:
This won’t work for the `media` option but it can be an acceptable compromise.

Out of curiosity, why don’t you just memoize the plugins array in your app?

Reply:
I originally thought it was supposed to work like wavesurfer.js, but it looks like it's better to use useMemo